app/src/main/java/org/schabi/newpipe/util/external_communication/TextLinkifier.java

```diff
@@ -0,0 +1,283 @@
+     * the content in the popup player at the time indicated in the timestamp, by using
+     * {@link TextLinkifier#addClickListenersOnTimestamps(Context, SpannableStringBuilder, Info,
+     * CompositeDisposable)} method and click listeners on hashtags, by using
+     * {@link TextLinkifier#addClickListenersOnHashtags(Context, SpannableStringBuilder, Info)},
+     * which will open a search on the current service with the hashtag.
+     * <p>
+     * This method is required in order to intercept links and e.g. show a confirmation dialog
+     * before opening a web link.
+     *
+     * @param textView    the TextView in which the converted CharSequence will be applied
+     * @param chars       the CharSequence to be parsed
+     * @param relatedInfo if given, handle timestamps to open the stream in the popup player at
+     *                    the specific time, and hashtags to search for the term in the correct
+     *                    service
+     * @param disposables disposables created by the method are added here and their lifecycle
+     *                    should be handled by the calling class
+     */
+    private static void changeIntentsOfDescriptionLinks(final TextView textView,
+                                                        final CharSequence chars,
+                                                        @Nullable final Info relatedInfo,
+                                                        final CompositeDisposable disposables) {
+        disposables.add(Single.fromCallable(() -> {
+            final Context context = textView.getContext();
+
+            // add custom click actions on web links
+            final SpannableStringBuilder textBlockLinked = new SpannableStringBuilder(chars);
+            final URLSpan[] urls = textBlockLinked.getSpans(0, chars.length(), URLSpan.class);
+
+            for (final URLSpan span : urls) {
+                final String url = span.getURL();
+                final ClickableSpan clickableSpan = new ClickableSpan() {
+                    public void onClick(@NonNull final View view) {
+                        if (!InternalUrlsHandler.handleUrlDescriptionTimestamp(
+                                new CompositeDisposable(), context, url)) {
+                            ShareUtils.openUrlInBrowser(context, url, false);
+                        }
+                    }
+                };
+
+                textBlockLinked.setSpan(clickableSpan, textBlockLinked.getSpanStart(span),
+                        textBlockLinked.getSpanEnd(span), textBlockLinked.getSpanFlags(span));
+                textBlockLinked.removeSpan(span);
+            }
+
+            // add click actions on plain text timestamps only for description of contents,
+            // unneeded for meta-info or other TextViews
+            if (relatedInfo != null) {
+                addClickListenersOnTimestamps(context, textBlockLinked, relatedInfo, disposables);
```

Comment:
Maybe this should be called only if `relatedInfo instanceof StreamInfo`? What do you think?

Reply:
I think it's a good idea, I will commit it.